lib/react_client/react_interop.dart

```diff
@@ -82,8 +82,10 @@ class Ref<T> {
   T get current {
     final jsCurrent = jsRef.current;
 
-    if (jsCurrent is! Element) {
-      final dartCurrent = (jsCurrent as ReactComponent)?.dartComponent;
+    // Note: this ReactComponent check will pass for many types of JS objects,
+    // so don't assume for sure that it's a ReactComponent
+    if (jsCurrent is! Element && jsCurrent is ReactComponent) {
```

Comment:
I added this `is ReactComponent` check since using the Dart wrapper with a non-component value threw (which may not much in practice, but is a total valid use of `createRef`).